mysql-test/suite/galera/t/pxc_component_keyring_file.test

```diff
@@ -0,0 +1,136 @@
+
+  --let $assert_text= Assert that the server has no keys created in the keyring
+  --let $assert_cond= [SELECT COUNT(*) = 0 from performance_schema.keyring_keys]
+  --source include/assert.inc
+
+  --connection node_2
+  --let $assert_text= Assert that the server has started with a keyring component
+  --let $assert_cond= [SELECT STATUS_VALUE = "Active" FROM performance_schema.keyring_component_status WHERE STATUS_KEY="Component_status"]
+  --source include/assert.inc
+
+  --let $assert_text= Assert that the server is component_keyring_file enabled
+  --let $assert_cond= [SELECT STATUS_VALUE = "component_keyring_file" FROM performance_schema.keyring_component_status WHERE STATUS_KEY="Implementation_name"]
+  --source include/assert.inc
+
+  --connection node_1
+  # Create an encrypted table
+  CREATE TABLE t1 (f1 INTEGER PRIMARY KEY) ENCRYPTION='Y';
+  INSERT INTO t1 VALUES (1),(2),(3),(4);
+
+  --let $assert_text= Assert that the server has created the master key on node_1
+  --let $assert_cond= [SELECT COUNT(*) = 1 from performance_schema.keyring_keys]
+  --source include/assert.inc
+
+  --connection node_2
+  # Joiner will have two keys. The one restored by PXB, and the other one by server.
+  --let $assert_text= Assert that the server has created the master key on node_2
+  --let $assert_cond= [SELECT COUNT(*) = 2 from performance_schema.keyring_keys]
+  --source include/assert.inc
+
+  SHOW CREATE TABLE t1;
+  SELECT * FROM t1;
+
+  ALTER INSTANCE ROTATE INNODB MASTER KEY;
+
+  --let $assert_text= Assert that the server has rotated the master key on node_2
+  --let $assert_cond= [SELECT COUNT(*) = 3 from performance_schema.keyring_keys]
+  --source include/assert.inc
+
+  --connection node_1
+  --let $assert_text= Assert that the server has rotated the master key on node_1
+  --let $assert_cond= [SELECT COUNT(*) = 2 from performance_schema.keyring_keys]
+
+  # Test with record set and gcache encryption
+  --let $restart_parameters = "restart:--wsrep-disk-pages-encrypt=ON --wsrep-gcache-encrypt=ON"
+  --source suite/galera/include/restart_cluster_with_sst.inc
+
+  # Check that GCache ring buffer file content is encrypted on node_1
+  --let $MYSQL_DATA_DIR = `SELECT @@datadir`
+  --let $assert_text = Assert that GCache is encrypted on node_1
+  --let $assert_file = $MYSQL_DATA_DIR/galera.cache
+  --let $assert_select = enc_encrypted: 1
+  --let $assert_count = 1
+  --source include/assert_grep.inc
+
+  # Check that GCache ring buffer file content is encrypted on node_2
+  --connection node_2
+  --let $MYSQL_DATA_DIR = `SELECT @@datadir`
+  --let $assert_text = Assert that GCache is encrypted on node_2
+  --let $assert_file = $MYSQL_DATA_DIR/galera.cache
+  --let $assert_select = enc_encrypted: 1
+  --let $assert_count = 1
+  --source include/assert_grep.inc
+
+  --let $restart_parameters = "restart"
+  --source suite/galera/include/restart_cluster_with_sst.inc
+
+  # Check that GCache ring buffer file content is not encrypted on node_1
```

Comment:
We should switch back to node_1

Reply:
We are already on `node_1`. The inc file `suite/galera/include/restart_cluster_with_sst.inc` swicthes to connection `node_1` in the end.

Reply:
I would add it anyway, to avoid the confusion

Reply:
Done